openapi-build-extension/src/main/java/io/ballerina/openapi/build/HttpServiceAnalysisTask.java

```diff
@@ -0,0 +1,131 @@
+        List<OASResult> openAPIDefinitions = new ArrayList<>();
+        for (Node node : modulePartNode.members()) {
+            SyntaxKind syntaxKind = node.kind();
+            // Load a service declarations
+            if (syntaxKind == SyntaxKind.SERVICE_DECLARATION) {
+                openAPIDefinitions.addAll(ServiceToOpenAPIConverterUtils.generateOAS3Definition(syntaxTree,
+                        semanticModel, null, false, inputPath));
+            }
+        }
+        List<Diagnostic> diagnostics = new ArrayList<>();
+        if (!openAPIDefinitions.isEmpty()) {
+            extractOpenAPIYamlFromOutputs(outPath, openAPIDefinitions, diagnostics);
+        }
+        if (!diagnostics.isEmpty()) {
+            for (Diagnostic diagnostic : diagnostics) {
+                context.reportDiagnostic(diagnostic);
+            }
+        }
+    }
+
+    private void extractOpenAPIYamlFromOutputs(Path outPath, List<OASResult> openAPIDefinitions,
+                                               List<Diagnostic> diagnostics) {
+        for (OASResult oasResult: openAPIDefinitions) {
+            if (oasResult.getYaml().isPresent()) {
+                try {
+//                    String fileName = checkDuplicateFiles(outPath, oasResult.getServiceName(), false);
+                    // Create openapi directory if not exists in the path. If exists do not throw an error
+                    Files.createDirectories(Paths.get(outPath + OAS_PATH_SEPARATOR + OPENAPI));
+                    String fileName = resolveContractFileName(outPath.resolve(OPENAPI), oasResult.getServiceName(),
+                            false);
+                    writeFile(outPath.resolve(OPENAPI + OAS_PATH_SEPARATOR + fileName), oasResult.getYaml().get());
+                } catch (IOException e) {
+                    DiagnosticMessages error = DiagnosticMessages.OAS_CONVERTOR_108;
+                    ExceptionDiagnostic diagnostic = new ExceptionDiagnostic(error.getCode(),
+                            error.getDescription(), null, e.toString());
+                    diagnostics.add(BuildExtensionUtil.getDiagnostics(diagnostic));
+                }
+            }
+            if (!oasResult.getDiagnostics().isEmpty()) {
+                for (OpenAPIConverterDiagnostic diagnostic: oasResult.getDiagnostics()) {
+                    diagnostics.add(BuildExtensionUtil.getDiagnostics(diagnostic));
+                }
+            }
+        }
+    }
+
+    /**
+     * Writes a file with content to specified {@code filePath}.
+     *
+     * @param filePath valid file path to write the content
+     * @param content  content of the file
+     * @throws IOException when a file operation fails
+     */
+    public static void writeFile(Path filePath, String content) throws IOException {
+        try (FileWriter writer = new FileWriter(filePath.toString(), StandardCharsets.UTF_8)) {
+            writer.write(content);
+        }
+    }
```

Comment:
Shall we move this method into the util class?

Reply:
Yeah , will move to separate util class